presto-tests/src/main/java/io/prestosql/tests/AbstractTestQueries.java

```diff
@@ -3311,9 +3311,12 @@ public void testCorrelatedScalarSubqueries()
         // two level of nesting
         assertQuery("SELECT * FROM nation n WHERE 2 = (SELECT (SELECT 2 * n.nationkey))");
 
+        // redundant LIMIT in subquery
+        assertQuery("SELECT (SELECT count(*) FROM (VALUES (7,1)) t(orderkey, value) WHERE orderkey = corr_key LIMIT 1) FROM (values 7) t(corr_key)");
```

Comment:
This does not test your rule, please also add similar test (but with plan verification) to `TestLogicalPlanner`

Reply:
Actually previously this kind of query failed but now it is not happening as the limit node is removed.